### ISSUE TITLE
Here are the steps to reduce big chunks of text - SRT

### ISSUE THREAD

**Issue by javaintheuk** [abhirooptalasila/AutoSub#21]
Hi,

I did this manually but maybe someone can improve it and write some script for it?:

1. Check text that is longer than 7 words

2. Add a line break to each line longer than 7 words.

3. Get the numbers of lines you got, ie: 
blah blah blah blah blah blah blah
blah blah blah blah blah blah blah
blah blah blah blah blah blah blah
blah blah blah blah blah

There are 4 lines
4. We take the initial and final time

13 <<< SRT Subtitle position
ie: 00:00:25,90 --> 00:00:35,25

5. We surplus them and divide them by the number of lines 
35,25 - 25,90 = 9,35
4 lines of max 7 words each
9,35/4 = 2,33

6. We take 2.33 less 1 for each limit, Ie:

13 <<< SRT Subtitle position
**00:00:25,90** --> 00:00:28,23
blah blah blah blah blah blah blah

14
00:00:28,24 --> 00:00:30,57
blah blah blah blah blah blah blah

15
00:00:30,58 --> 00:00:32,91
blah blah blah blah blah blah blah

16
00:00:32,92 --> **00:00:35,25**
blah blah blah blah blah

14 <<<<  WARNING  >>>> update all the other numbers in this case to 17 (check step 7 below)
blah blah blah blah

7.  We need to update the SRT Subtitle position, in this case we finished at 16, so we need to replace 14 for 17, and do the same for all the other numbers. Note: Update from top to bottom so the counter increments.

That's it.

Anyone? :)

**Comment by abhirooptalasila** [comment]
Sounds like a decent workaround. Can you raise a PR?

**Comment by abnerLing** [comment]
Instead of looking at the text to reduce chunks would it be possible to modify the segmentAudio.py script to split with a max duration? I see that there is a line (below) that removes segments based on a minimum duration. I'm not too sure how that would look but I presume we could have a variable such as max_duration = 10 or something like that...

```
    min_duration = 0.2
    seg_limits_2 = []
    for s_lim in seg_limits:
        if s_lim[1] - s_lim[0] > min_duration:
            seg_limits_2.append(s_lim)
    seg_limits = seg_limits_2
```

**Comment by shasheene** [comment]
OK I have fixed this issue and added `--split-duration` command-line option.

With my fix, run-on sentences are automatically get split up into multiple smaller subtitles, based on duration as @abnerLing suggested.

If `--split-duration` is not specified, it uses what I deem to be a sensible default (5 seconds).

Note: No changes have been done to the splitting of the input audio. This means the inferred transcript is identical to before my change, it's just how the subtitles are being written out that's changed.

Also I have confirmed the VTT cue timings continue to be correctly handled.

This change is part of #32. Once it has been merged, everybody can enjoy the improvements :)

I hope to continue to be an active developer/maintainer on this promising project :)
